src/legacy/ui/public/legacy_compat/ensure_default_index_pattern.tsx

```diff
@@ -37,31 +37,41 @@ let timeoutId: NodeJS.Timeout | undefined;
  * If there are no index patterns, redirect to management page and show
  * banner. In this case the promise returned from this function will never
  * resolve to wait for the URL change to happen.
+ *
+ * An existing default index pattern id is returned.
+ * Given persist is set to true, a new defined defaultIndex is written to uiSettings
  */
 export async function ensureDefaultIndexPattern(
   newPlatform: CoreStart,
   data: DataPublicPluginStart,
   $rootScope: IRootScopeService,
-  kbnUrl: any
+  kbnUrl: any,
+  persist: boolean = true
```

Comment:
nit: Can we put this into an `options` object? It's a personal taste, but I think that pattern is much better readable than simple boolean arguments.

Reply:
was generally refactored so this part of the code wasn't modified in this PR